db.go

```diff
@@ -230,28 +239,57 @@ func (db *DB) OpenSSTable(file *os.File) (*SSTable, error) {
 	return sst, nil
 }
 
-func OpenSSTable(filename string) (*SSTable, error) {
-	file, err := os.Open(filename)
-	if err != nil {
-		return nil, err
+// todo: can be improved
+func (db *DB) deleteSStableAtLevel(level int, iterators []*SSTableIterator) {
+	iteratorMap := make(map[*SSTable]bool)
+	for _, iterator := range iterators {
+		iteratorMap[iterator.sstable] = true
 	}
-	sst := NewSSTable(file)
-	return sst, nil
+
+	newSSTables := make([]*SSTable, 0)
```

Comment:
### Security Issues
Ensure that the `db.deleteSStableAtLevel(level, iterators)` method securely deletes the SSTables. Use `os.RemoveAll` instead of `os.Remove` to ensure that all files and directories are securely deleted. Additionally, consider adding a check to ensure that the files being deleted are indeed SSTables and not arbitrary files.